pkg/cmd/pulumi/deployment_settings_config.go

```diff
@@ -327,32 +328,102 @@ func newDeploymentSettingsConfigureCmd() *cobra.Command {
 	return cmd
 }
 
+func newRepoLookup(repoRoot string) (repoLookup, error) {
+	if repoRoot != "" {
+
+		repo, err := git.PlainOpen(repoRoot)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
Could we do something like the following here and then we wouldn't even need the `detectGitRootDirectory` function?

```suggestion
		repo, err := git.PlainOpen(repoRoot)
		if err == git.ErrRepositoryNotExists {
			return  &noRepoLookupImpl{}, nil
		}
		if err != nil {
			return nil, err
		}
```

Reply:
much cleaner now, done